internal/sys/fs.go

```diff
@@ -169,6 +169,15 @@ func (r *lazyDir) Stat() (platform.Stat_t, syscall.Errno) {
 	}
 }
 
+// Chown implements the same method as documented on platform.File
+func (r *lazyDir) Chown(uid, gid int) syscall.Errno {
+	if f, err := r.file(); err != 0 {
+		return err
```

Comment:
the logic seems not in line with `File()`:

```go
	if f, err := r.file(); err != 0 {
		panic(err) // Bad, but temporary
	}
```

Reply:
so `lazyDir` is a `platform.File`, so it can return a `syscall.Errno` (the second result of `r.file()`).

I guess you mean there's a generally problematic thing about error mapping because the implementation of `r.file()` opens the file lazily. If it is later missing, I think we should return EBADF not panic, right? as that's normal for things like this. So maybe panic if something besides ENOENT?

Reply:
ah ok that makes sense and I'm good with as-is. Thank you for clarifying!

Reply:
well it is half ok.. we are leaking more errnos than ENOENT, so I need to fix it. Good catch

Reply:
I made the panic more precise. Good eye!